### PR TITLE
fix: Code scanning issues

### PR DESCRIPTION
**Pull request by JeyJeyGao**
Fix:
- Token-Permissions issue for dev-release.yml: delete the file
- Token-Permissions issue for license-checker.yml: update the global permission to be read

Resolves part of #408 
Signed-off-by: Junjie Gao <junjiegao@microsoft.com>

### CODE REVIEW

File: .github/workflows/dev-release.yml
[the remark this answers is not in the document]
Comment (JeyJeyGao) [reply .github/workflows/dev-release.yml:1]: The dev-release is for a weekly release during the Alpha phase, so we don't need it now. The file violates the token-permission rule, so I removed it.

**Review by Two-Hearts — APPROVED** [review]
LGTM

**Review by shizhMSFT — APPROVED** [review]
LGTM. Glad to see we deprecate the dev release which is not used for a long time.

**Review by priteshbandi — APPROVED** [review]
LGTM

**Review by priteshbandi — APPROVED** [review]
LGTM

**Review by Two-Hearts — APPROVED** [review]
LGTM

**Pull request merged**